infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -300,18 +304,17 @@ private void connectToClusterManager() {
      * Both:
      * - Metadata Managers which maintain metadata related to replication and its status
      */
-    private void performRoleBasedSetup(TopologyDescriptor topology) {
-
-        ClusterRole role = topology.getLocalClusterDescriptor().getRole();
+    private void performRoleBasedSetup() {
 
-        if (role != ClusterRole.SOURCE && role != ClusterRole.SINK) {
-            log.debug("Cluster role is {}.  Not performing role-based setup.", role);
+        if (!isSource() && !isSink()) {
+            log.debug("Cluster is neither SOURCE nor SINK.  Not performing role-based setup.");
             return;
         }
 
-        sessionManager.refresh(topologyDescriptor);
+        sessionManager.refresh(topologyDescriptor, fetchConnectionEndpoints());
 
-        if (role == ClusterRole.SOURCE) {
+        // Check if local cluster is a SOURCE
+        if (isSource()) {
```

Comment:
That change will come in with the connection model PR as it is coupled with the connection Model.

Decoupling it would be a little time consuming and redundant. So this PR is continuing to be for FULL_TABLE model only, but is consuming the new clusterManager APIs.